base/reduce.jl

```diff
@@ -1246,3 +1184,64 @@ function _simple_count(::typeof(identity), x::Array{Bool}, init::T=0) where {T}
     end
     return n
 end
+
+# simd optimization for min/max related reduction
+# general fallback
+_fast(op, x, y) = op(x, y)
+_fast(op) = (x, y) -> _fast(op, x, y)
+
+# The following are used in optimized `mapreduce_impl` for IEEEFloat.
+_fast(::typeof(min), x::T, y::T) where {T<:IEEEFloat} = ifelse(signbit(x-y), x, y)
+_fast(::typeof(max), x::T, y::T) where {T<:IEEEFloat} = ifelse(signbit(y-x), x, y)
+
+# Some help function for better vectorization
+# NaN check
+_anynan(x, y) = isnan(x) | isnan(y)
+_anynan(x::NTuple{N}) where {N} = reduce(|, map(_anynan, x[1:N>>1], x[1+N>>1:N]))
+# split then reduce for inputs whose length > 4
+_half_reduce(op, x::NTuple{N}) where {N} = _half_reduce(op, map(op, x[1:N>>1], x[1+N>>1:N]))
+_half_reduce(op, x::NTuple{4}) = reduce(op, x)
+
+function mapreduce_impl(f, op::Union{typeof(max),typeof(min)},
+                        A::AbstractArrayOrBroadcasted, first::Int, last::Int)
+    T = @default_eltype A
+    Eltype = _return_type(f, Tuple{T})
+    # Limit this optimization to concrete IEEEFloat. (general fallback is 2x faster for Integer)
+    (isconcretetype(Eltype) && Eltype <: IEEEFloat) ||
+        return mapreduce_impl(f, op, A, first, last, pairwise_blocksize(f, op))
+    @inline elf(i::Int) = @inbounds f(A[i])
+    @noinline firstnan(temp::Tuple) = temp[findfirst(isnan, temp)], last
+    function simd_kernal(::Val{N}, ini) where {N}
```

Comment:
`simd_kernal` should probably read `simd_kernel`

Reply:
nice catch.